File: app/src/main/java/org/dhis2/utils/RulesUtilsProvider.java
package org.dhis2.utils;

import android.support.annotation.NonNull;

import org.dhis2.data.forms.dataentry.fields.FieldViewModel;
import org.hisp.dhis.android.core.program.ProgramStageModel;
import org.hisp.dhis.rules.models.RuleEffect;

import java.util.Map;

/**
 * QUADRAM. Created by ppajuelo on 13/06/2018.
 */

public interface RulesUtilsProvider {


    void applyRuleEffects(Map<String, FieldViewModel> fieldViewModels, Result<RuleEffect> calcResult,
                          @NonNull RulesActionCallbacks rulesActionCallbacks);

    void applyRuleEffects(Map<String, ProgramStageModel> programStages, Result<RuleEffect> calcResult);

}

File: app/src/main/java/org/dhis2/utils/RulesUtilsProvider.kt
package org.dhis2.utils

import org.dhis2.data.forms.dataentry.ValueStore
import org.dhis2.form.model.FieldUiModel
import org.hisp.dhis.android.core.program.ProgramStage
import org.hisp.dhis.rules.models.RuleEffect

interface RulesUtilsProvider {

    fun applyRuleEffects(
        applyForEvent: Boolean,
        fieldViewModels: MutableMap<String, FieldUiModel>,
        calcResult: Result<RuleEffect>,
        valueStore: ValueStore?
    ): RuleUtilsProviderResult

    fun applyRuleEffects(
        programStages: MutableMap<String, ProgramStage>,
        calcResult: Result<RuleEffect>
    )
}
